README.md

```diff
@@ -5,3 +5,312 @@
 
 This repository contains the Python client SDK for [Eclipse Ditto](https://eclipse.org/ditto/).
 
+## Table of Contents
+* [Installation](#Installation)
+* [Creating and connecting a client](#Creating-and-connecting-a-client)
+  * [Creating a client instance as a class](#Creating-a-client-instance-as-a-class)
+  * [Connecting an external paho client](#Connecting-an-external-paho-client)
+* [Working with features](#Working-with-features)
+  * [Creating a new feature instance](#Creating-a-new-feature-instance)
+  * [Modifying a feature's property](#Modifying-a-feature's-property)
+  * [Deleting a feature's property](#Deleting-a-feature's-property)
+  * [Deleting a feature](#Deleting-a-feature)
+* [Subscribing and handling messages](#Subscribing-and-handling-messages)
+* [Logging](#Logging)
+
+## Installation
+
+Install the sources and execute the following command:
+
+```commandline
+make install
+```
+
+## Creating and connecting a client
+
+Each client instance can optionally have a defined behaviour after connecting to or disconnecting from the client.
+
+```python    
+def on_connect(ditto_client: Client):
+    print("Ditto client connected")
+
+def on_disconnect(ditto_client: Client):
+    print("Ditto client disconnected")
+```
+
+With these configurations a client instance can be created.
+
+```python
+client = Client(on_connect=on_connect, on_disconnect=on_disconnect)
+```
+
+After the client is created, it's ready to be connected.
+
+```python
+client.connect("localhost")
+while True:
+    try:
+        time.sleep(5)
+    except KeyboardInterrupt:
+        print("finished")
+        client.disconnect()
+        sys.exit()
+```
+
+Full example of the basic client connection can be found [here](examples/client_connect.py).
+
+### Creating a client instance as a class
```

Comment:
To keep the README crisp - this section (class implementation approach) could be just mentioned as a usage possibility with a reference to the full example but not as a whole section.

The key info in this example is actually the possibility to use an external Paho client.